pkgs/build-support/node/build-npm-package/hooks/npm-install-hook.sh

```diff
@@ -43,21 +43,6 @@ npmInstallHook() {
     local -r nodeModulesPath="$packageOut/node_modules"
 
     if [ ! -d "$nodeModulesPath" ]; then
-        if [ -z "${dontNpmPrune-}" ]; then
-            if ! npm prune --omit=dev --no-save ${npmWorkspace+--workspace=$npmWorkspace} $npmPruneFlags "${npmPruneFlagsArray[@]}" $npmFlags "${npmFlagsArray[@]}"; then
-              echo
-              echo
-              echo "ERROR: npm prune step failed"
-              echo
-              echo 'If npm tried to download additional dependencies above, try setting `dontNpmPrune = true`.'
-              echo
-
-              exit 1
-            fi
-        fi
-
-        find node_modules -maxdepth 1 -type d -empty -delete
-
         cp -r node_modules "$nodeModulesPath"
```

Comment:
Hmm this is a bit worrying - The `installPhase` will run before the `fixupPhase`, and will copy this directory before the fixup happened.. Then, if you won't run `npm prune`, you will always get all of the development dependencies in the `$packageOut/node_modules`... That's probably the reason why the author of this hook saw this part as an nondetachable part of the installation phase.